docs/key-constraints.rst

```diff
@@ -0,0 +1,44 @@
+.. _ksql_key_constraints:
+
+============
+Message Keys
+============
+
+The `CREATE STREAM` and `CREATE TABLE` statements allow the user to specify the column that corresponds to the Kafka message key by setting the `KEY` property of the `WITH` clause. For example:
+
+.. code:: sql
+
+    CREATE TABLE users (registertime bigint, gender varchar, regionid varchar, userid varchar) WITH (KAFKA_TOPIC=‘users', VALUE_FORMAT=‘JSON', KEY = 'userid');
+
+
+In the case of tables, the `KEY` property is required.
+
+In the case of streams, the `KEY` property is optional. KSQL uses it as a hint to determine if a repartition can be avoided when performing aggregations and joins.
+
+In either case, when setting `KEY` the user must be sure that the following is true:
+
+    - `KEY` must be set to a column of type `STRING` or `VARCHAR`.
+    - The value in the message key must be the same as the value in the column set in `KEY` for every record.
```

Comment:
Add a section "What To Do If Your Topic has no key, but the key is a field in the value structure"